chore: filters traditional vs agentic rag (#318)
agno-agi/agno-docs: filters/introduction.mdx

> **Tip:** You can combine multiple filters for more precise results!

## Filters in Traditional RAG vs. Agentic RAG

> When configuring your Agent, it's important to choose the right approach for your use case. In traditional RAG, you typically set `add_references=True` to ensure that references are included in the responses. However, in Agentic RAG, you should use `search_knowledge=True` to leverage the agent's ability to infer and apply filters automatically. 

Reviewer: ```suggestion
When configuring your Agent it is important to choose the right approach for your use case. There are two broad approaches to RAG with Agno agents: traditional RAG and agentic RAG. With a traditional RAG approach youset `add_references=True` to ensure that references are included in the responses. For Agentic RAG, you set `search_knowledge=True` to leverage the agent's ability to infer and apply filters automatically. 
```

Example:
```python
agent = Agent(
    name="KnowledgeFilterAgent",
    search_knowledge=False,  # Do not use agentic filtering

>     add_references=True,     # Useuse manual filtering

Reviewer: ```suggestion
    add_references=True,     # Add knowledge base references to the system prompt
```

Should you not pass filters? 

Also we should probably add an example.

> **Tip:** You can combine multiple filters for more precise results!

## Filters in Traditional RAG vs. Agentic RAG

When configuring your Agent it is important to choose the right approach for your use case. There are two broad approaches to RAG with Agno agents: traditional RAG and agentic RAG. With a traditional RAG approach youset `add_references=True` to ensure that references are included in the responses. For Agentic RAG, you set `search_knowledge=True` to leverage the agent's ability to infer and apply filters automatically. 

Example:
```python
agent = Agent(
    name="KnowledgeFilterAgent",

>     search_knowledge=False,  # Do not use agentic filtering

Reviewer: ```suggestion
    search_knowledge=False,  # Do not use agentic search
```
